### PR TITLE
Upgrades to Apollos 1.7

### PR DESCRIPTION
**Pull request by redreceipt**
## DESCRIPTION

Screen Shot 2020-10-21 at 5 14 52 PM

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/WillowCreekApp/src/content-single/ContentSingle.js [hunk 1]
```diff
@@ -41,10 +40,12 @@ class ContentSingle extends PureComponent {
   }
 
   renderContent = ({ content, loading, error }) => {
+    console.log(content);
```
Comment (vinnyjth) [inline packages/WillowCreekApp/src/content-single/ContentSingle.js:43]:
```suggestion
```

File: packages/WillowCreekApp/src/content-single/ContentSingle.js [hunk 2]
```diff
@@ -41,10 +40,12 @@ class ContentSingle extends PureComponent {
...
     let { __typename } = content;
     if (!__typename && this.itemId) {
       [__typename] = this.itemId.split(':');
     }
+    console.log(__typename);
```
Comment (vinnyjth) [inline packages/WillowCreekApp/src/content-single/ContentSingle.js:48]:
```suggestion
```

File: packages/WillowCreekApp/src/content-single/WeekendContentItem/index.js [hunk 3]
```diff
@@ -95,13 +96,14 @@ const WeekendContentItem = ({ content, loading }) => {
                     <H2 padded isLoading={!content.title && loading}>
                       {content.title}
                     </H2>
+                    <ContentHTMLViewConnected contentId={content.id} />
+                    <StyledContentHTMLViewConnected contentId={content.id} />
```
Comment (vinnyjth) [inline packages/WillowCreekApp/src/content-single/WeekendContentItem/index.js:100]: I'm assuming we don't need both of these?

File: packages/WillowCreekApp/src/user-settings/Locations/index.js [hunk 4]
```diff
@@ -22,10 +21,7 @@ const Location = (props) => (
         }}
         changeCampusOptions={{
           awaitRefetchQueries: true,
-          refetchQueries: [
-            { query: GET_FEED_FEATURES, variables: undefined },
-            { query: GET_CONTENT_CHANNELS, variables: undefined },
-          ],
+          refetchQueries: [{ query: GET_FEED_FEATURES, variables: undefined }],
```
Comment (vinnyjth) [inline packages/WillowCreekApp/src/user-settings/Locations/index.js:24]: This might need to be updated to refer to the `GET_HOME_FEED` query

**Comment by redreceipt** [comment]
blocked on node media query making through on a new `ui-connected` alpha

File: packages/WillowCreekApp/src/user-settings/Locations/index.js (hunk 4 above)
Comment (redreceipt) [reply packages/WillowCreekApp/src/user-settings/Locations/index.js:24]: good catch

Comment (redreceipt) [inline packages/WillowCreekApp/src/user-settings/Locations/index.js:24]:
```suggestion
          refetchQueries: [{ query: GET_HOME_FEED, variables: undefined }],
```

File: packages/WillowCreekApp/src/content-single/ContentSeriesContentItem/index.js [hunk 5]
```diff
@@ -36,6 +36,7 @@ const Content = styled(() => ({
 }))(View);
 
 const ContentSeriesContentItem = ({ content, loading }) => {
+  console.log(content);
```
Comment (vinnyjth) [inline packages/WillowCreekApp/src/content-single/ContentSeriesContentItem/index.js:39]:
```suggestion
```

**Comment by redreceipt** [comment]
ok this one should be ready again

**Comment by vinnyjth** [comment]
I was testing this last night, I was running into issues with the Home Feed not refreshing properly after changing campuses. I tracked it down to FeatureIds not being different per campus. I'm close to solving it locally, and we should be good after that.

**Comment by redreceipt** [comment]
o shoot, thanks. I thought I saw them changing. where did I go wrong?

**Comment by vinnyjth** [comment]
I logged campusId in `getFeatureId`, and it was undefined. I think you needed to put the actual `campusId` into the context. I forgot to push my changes from yesterday at home, so hold on as I call my wife and have her push code for me 😂

**Comment by vinnyjth** [comment]
@redreceipt Like dis: https://github.com/Differential/willow-creek/pull/154/commits/a0b1d87f1e7e7078415755e4db41e47699028002#diff-aeced4b46c2c226bad68cfed9edb9873083abe28a40eda0fc85cd50469af5893R44

File: packages/apollos-api/src/data/feature-feeds/index.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+class FeatureFeed extends coreFeatureFeed.dataSource {
+  baseGetFeed = this.getFeed;
+
+  // this will also embed campus into the feature feed id
+  getFeed = async ({ type, args }) => {
+    const {
+      campusId,
+    } = await this.context.dataSources.Person.getCurrentUserCampusId();
+    return this.baseGetFeed({ type, args: { ...args, campusId } });
```
Comment (redreceipt) [inline packages/apollos-api/src/data/feature-feeds/index.js:13]: @vinnyjth thought i was doing that here

Comment (vinnyjth) [reply packages/apollos-api/src/data/feature-feeds/index.js:13]: You had the code perfect for the `FeedIds`, but the features themselves all shared the same IDs. Meant that the feed itself would rerender, but the Features would each have the same ids as before and thus wouldn't decide to rerender.

File: packages/apollos-api/src/data/features/data-source.js [hunk 7]
```diff
@@ -77,14 +77,24 @@ export default class Feature extends baseFeature.dataSource {
...
+      ...args,
+    });
   }
 
+  // pop out campus ID from feature feed id
+  baseGetFeatures = this.getFeatures;
+
+  getFeatures = async (args) => {
+    // campusId is only in the args from our custom FeatureFeed.getFeed function
+    this.context.campusId = args.campusId;
```
Comment (redreceipt) [inline packages/apollos-api/src/data/features/data-source.js:94]: and then it was getting loaded in here

Comment (vinnyjth) [reply packages/apollos-api/src/data/features/data-source.js:94]: The campusId `arg` isn't passed through I think. `args` is an array of features (the config). We need to change this is core so the second arg of `getFeatures` is the `args` for the FeatureFeed

File: packages/apollos-api/src/data/feature-feeds/index.js (hunk 6 above)
Comment (redreceipt) [reply packages/apollos-api/src/data/feature-feeds/index.js:13]: oh like if you changed the campus?

Comment (vinnyjth) [reply packages/apollos-api/src/data/feature-feeds/index.js:13]: Yah, totally. Worked fine in all other situations.

File: packages/apollos-api/src/data/features/data-source.js (hunk 7 above)
Comment (redreceipt) [reply packages/apollos-api/src/data/features/data-source.js:94]: oh I see. I meant to embed campus id one level deeper on each feature config

File: packages/apollos-api/src/data/features/resolver.js [hunk 8]
```diff
@@ -32,8 +32,18 @@ function getSupportedTypes(info) {
...
+    // deprecated
+    userFeedFeatures: async (
+      root,
+      args,
+      { dataSources: { Feature, Person } },
+      info
+    ) => {
+      // pops currentPerson into the context
+      const { campusId } = await Person.getCurrentUserCampusId();
+      this.context.campusId = campusId;
```
Comment (redreceipt) [inline packages/apollos-api/src/data/features/resolver.js:44]: this is just for backward compatibility correct?

Comment (vinnyjth) [reply packages/apollos-api/src/data/features/resolver.js:44]: totes.

**Review by vinnyjth — APPROVED** [review]
